env.d/60-flatpak

```diff
@@ -0,0 +1,15 @@
+# flatpak-environment-generator
+
+#!/bin/bash
+
+if [ "${XDG_DATA_DIRS#*flatpak}" = "${XDG_DATA_DIRS}" ]; then
+    OLD_XDG_DATA_DIRS=${XDG_DATA_DIRS}
+    XDG_DATA_DIRS="${XDG_DATA_HOME:-"$HOME/.local/share"}/flatpak/exports/share"
+    while read install_path
+    do
+        XDG_DATA_DIRS="${XDG_DATA_DIRS}:$install_path/exports/share"
+    done < <(flatpak --list-installations)
```

Comment:
If you use

```
flatpak --list-installations | while read -r install_path
```

then I think this could be run by any POSIX shell (a `#!/bin/sh` script), which will reduce this script's impact on login time on distributions where `/bin/sh` is a smaller/faster shell (in particular, on Debian/Ubuntu it's `dash`).

Reply:
You realise that this doesn't work though? It's run in a separate shell, so $install_path isn't actually set when you need it.